core/src/swarm/registry.rs

```diff
@@ -53,7 +56,8 @@ impl Addresses {
     pub fn new(limit: NonZeroUsize) -> Self {
         Addresses {
             limit,
-            registry: SmallVec::new()
+            registry: SmallVec::new(),
+            reports: VecDeque::with_capacity(200),
```

Comment:
Instead of hard-coding the max. report length which has implications w.r.t. some use cases (as discussed) I suggest to take it as an additional parameter and use the 200 in the `Default` impl.